src/main/java/org/testng/internal/reflect/ReflectionRecipes.java

```diff
@@ -0,0 +1,411 @@
+package org.testng.internal.reflect;
+
+import org.testng.ITestContext;
+import org.testng.ITestResult;
+import org.testng.annotations.NoInjection;
+import org.testng.xml.XmlTest;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+
+/**
+ * Utility class to handle reflection.
+ *
+ * @author <a href="mailto:nitin.matrix@gmail.com">Nitin Verma</a>
+ */
+public final class ReflectionRecipes {
+
+  private static final Map<Class, Class> PRIMITIVE_MAPPING = new HashMap<Class, Class>();
+
+  static {
+    PRIMITIVE_MAPPING.put(boolean.class, Boolean.class);
+    PRIMITIVE_MAPPING.put(byte.class, Byte.class);
+    PRIMITIVE_MAPPING.put(short.class, Short.class);
+    PRIMITIVE_MAPPING.put(int.class, Integer.class);
+    PRIMITIVE_MAPPING.put(long.class, Long.class);
+    PRIMITIVE_MAPPING.put(float.class, Float.class);
+    PRIMITIVE_MAPPING.put(double.class, Double.class);
+    PRIMITIVE_MAPPING.put(char.class, Character.class);
+    PRIMITIVE_MAPPING.put(void.class, Void.class);
+  }
+
+  private ReflectionRecipes() {
+    throw new RuntimeException();
+  }
+
+  /**
+   * Checks if an instance is an instance of the given class.
+   *
+   * @param clazz  reference class.
+   * @param object instance to be tested.
+   * @return is an instance-of or not
+   */
+  public static boolean isInstanceOf(final Class clazz, final Object object) {
```

Comment:
No need for `final` here (same for local variables). Just a personal preference, I think `final` doesn't buy you much in these locations and it makes the code a lot more verbose.